editortask.cpp

```diff
@@ -444,14 +444,21 @@ void EditorTask::menuRun()
 {
     if(filepath == "")
     {
-        dialog_task.showMessage("You have to save the file first!");
-        return;
-    }
-
-    const char* argv[] = {filepath.c_str()};
-    key_hold_down = true;
-
-    nl_exec(settings_task.mpython_path.content().c_str(), sizeof(argv), const_cast<char**>(argv));
+		FILE *f = fopen("/documents/tmp.py", "wb");
+		if(f)
+		{
+			fwrite(buffer.c_str(), buffer.length(), 1, f);
+			fclose(f);
+			
+			runFile("/documents/tmp.py");
+			remove("/documents/tmp.py");
+		} else {
+			dialog_task.showMessage("Could not save 'tmp.py'!");
+		}
+
+    } else {
```

Comment:
Inconsistent style of bracing - here you can remove the {} completely, it's just a single statement